src/Features/Core/Portable/QuickInfo/QuickInfoContext.cs

```diff
@@ -14,30 +18,89 @@ internal sealed class QuickInfoContext
     {
         /// <summary>
         /// The document that quick info was requested within.
+        /// Can be <see langword="null"/> for quick info scenarios based off a <see cref="Compilation"/>,
+        /// where <see cref="SemanticModel"/> is guaranteed to be non-null.
         /// </summary>
-        public Document Document { get; }
+        public Document? Document { get; }
```

Comment:
goign to have to noodle on this.  my visceral reaction is extreme dislike.

Reply:
See https://github.com/dotnet/roslyn/pull/50334#discussion_r556763307. I agree that document should be guaranteed to be non-null for any public context API that we expose in future.